components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -190,42 +190,63 @@ public void setUp() throws Exception {
         messageContext2.setAuthorizedUser(user2);
 
         return new Object[][]{
-                {messageContext, tokenRespDTO, authzMessageContext, authzTokenRespDTO},
-                {messageContext2, tokenRespDTO, authzMessageContext, authzTokenRespDTO}
+                {messageContext, tokenRespDTO},
+                {messageContext2, tokenRespDTO}
         };
     }
 
     @Test(dataProvider = "TestBuildIDToken")
     public void testBuildIDToken(OAuthTokenReqMessageContext messageContext,
-                                 OAuth2AccessTokenRespDTO tokenRespDTO,
-                                 OAuthAuthzReqMessageContext authzMessageContext,
-                                 OAuth2AuthorizeRespDTO authzTokenRespDTO) throws Exception {
+                                 OAuth2AccessTokenRespDTO tokenRespDTO) throws Exception {
+
         RealmService realmService = IdentityTenantUtil.getRealmService();
         PrivilegedCarbonContext.getThreadLocalCarbonContext()
                 .setUserRealm(realmService.getTenantUserRealm(SUPER_TENANT_ID));
         IdpMgtServiceComponentHolder.getInstance().setRealmService(IdentityTenantUtil.getRealmService());
-        defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
-        defaultIDTokenBuilder.buildIDToken(authzMessageContext, authzTokenRespDTO);
+        String idToken = defaultIDTokenBuilder.buildIDToken(messageContext, tokenRespDTO);
+        Object o = SignedJWT.parse(idToken).getJWTClaimsSet();
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getAudience().get(0),
+                TestConstants.CLIENT_ID);
+        Assert.assertEquals(SignedJWT.parse(idToken).getJWTClaimsSet().getIssuer(),
```

Comment:
what is the difference between messageContext and messageContext2. I couldn't see any difference between them. 
You can configure different clientId , username in the OAuthTokenReqMessageContext and assert the clientId and username  from the idtoken